File: Aigilas/Aigilas/Entities/Extensions.cs
using Aigilas.Creatures;
using SPX.Entities;
using SPX.Core;
using System.Collections.Generic;

namespace Aigilas.Entities
{
    public static class Extensions
    {
        public static ICreature IsCreature(this IEntity entity)
        {
            if (entity.GetEntityType() == Aigilas.EntityType.ACTOR)
            {
                return entity as ICreature;
            }
            return null;
        }
    }
}

File: aigilas/src/aigilas/entities/Extensions.java
package aigilas.entities;

import aigilas.creatures.BaseCreature;
import sps.bridge.EntityTypes;
import sps.core.Core;
import sps.entities.Entity;

public class Extensions {
    public static BaseCreature isCreature(Entity entity) {
        if (entity.getEntityType() == EntityTypes.get(Core.Actor)) {
            return (BaseCreature) entity;
        }
        return null;
    }
}
